rollup/transaction_ingestion.go

```diff
@@ -129,6 +131,11 @@ func (t *TxIngestion) loop() {
 		for i, tx := range txs {
 			log.Debug("Transaction Ingestion", "hash", tx.Hash().Hex(), "submission index", index, "element", i)
 
+			// Set the timestamp of the chain equal to the timestamp from the
+			// L1 to L2 transaction
+			timestamp := timestamps[i]
+			t.bc.SetCurrentTimestamp(int64(timestamp))
```

Comment:
Just to make sure, we shouldn't have a race condition here at all right?

Reply:
We should add a check that makes sure the timestamp never goes down to this and the other timestamp update logic that we have